### PR TITLE
Add Links throughout learn course

### PR DESCRIPTION
**Pull request by eKhattak**
This PR is a fix for https://github.com/vercel/next.js/issues/15353 opened by @lfades 

Fixes https://github.com/vercel/next.js/issues/15353
Fixes https://github.com/vercel/next.js/issues/16011

### CODE REVIEW

File: pages/learn/basics/api-routes/index.mdx [hunk 1]
```diff
@@ -7,13 +7,18 @@ export const meta = {
   lessonId: 'api-routes'
 }
 
-Next.js has support for API Routes, which lets you easily create an API endpoint as a Node.js function. Although it’s not necessary for our blog app, we’ll briefly talk about how to use it in this lesson.
+<Link href="/docs/[[...slug]]" as="/docs/basic-features/built-in-css-support">
+  <a>CSS</a>
+</Link>
```
Comment (lfades) [inline pages/learn/basics/api-routes/index.mdx:12]: This change looks unintended:

image

File: pages/learn/basics/dynamic-routes/index.mdx [hunk 2]
```diff
@@ -13,11 +13,11 @@ We’ve populated the index page with the blog data, but we still haven’t crea
 
 In this lesson, you’ll learn:
 
-- How to statically generate pages with dynamic routes using `getStaticPaths`.
+- How to statically generate pages with dynamic routes using <Link href="/docs/[[...slug]]" as="/docs/basic-features/data-fetching#getstaticprops-static-generation"><a>`getStaticPaths`</a></Link>.
```
Comment (lfades) [inline pages/learn/basics/dynamic-routes/index.mdx:16]:
```suggestion
- How to statically generate pages with dynamic routes using <Link href="/docs/[[...slug]]" as="/docs/basic-features/data-fetching#getstaticpaths-static-generation"><a>`getStaticPaths`</a></Link>.
```

**Comment by eKhattak** [comment]
> @eKhattak Thank you! there are still some other cases where I would like to link but I'll do those myself, thanks!.

Can you point out those cases, I will it.

**Comment by lfades** [comment]
@eKhattak Any reference to anything that has a page in the Next.js docs, link to it, e.g if there is a mention to `next/link`, it should link to the next/link docs, same for any method like `useRouter`, `getStaticProps`, e.t.c, there are still many missing cases.

**Comment by eKhattak** [comment]
> @eKhattak Any reference to anything that has a page in the Next.js docs, link to it, e.g if there is a mention to `next/link`, it should link to the next/link docs, same for any method like `useRouter`, `getStaticProps`, e.t.c, there are still many missing cases.

Let me update it.

**Comment by eKhattak** [comment]
@lfades  Should I update all the links from `<Link ...` to `[content to show](link to visit)` ?

**Pull request merged**